R/CibersortX.R

```diff
@@ -176,23 +191,34 @@ deconvolute_cibersortx <- function(bulk_gene_expression, signature, verbose = FA
   if (is.null(signature)) {
     stop("Parameter 'signature' is missing or null, but it is required.")
   }
-  if (!docker_available()) {
-    message(
-      "Installation of docker can not be found. Please check whether you can ",
-      "call 'docker' in the command line and get a help menu"
-    )
-    return(NULL)
-  }
-  if (!docker_connectable()) {
-    message(
-      "Error durching connection to docker. Please check whether you can ",
-      "call 'docker ps' in the command line and get a (possibly empty) list and not an error ",
-      "message"
-    )
-    return(NULL)
+
+  if (container == "docker"){
+    if (!docker_available()) {
+      message(
+        "Installation of docker can not be found. Please check whether you can ",
+        "call 'docker' in the command line and get a help menu"
+      )
+      return(NULL)
+    }
+    if (!docker_connectable()) {
+      message(
+        "Error durching connection to docker. Please check whether you can ",
+        "call 'docker ps' in the command line and get a (possibly empty) list and not an error ",
+        "message"
+      )
+      return(NULL)
+    }
```

Comment:
since all of this is redundant (and now more redundant code is added) maybe move this to a helper function `setup_container()` or similar.